app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt

```diff
@@ -70,14 +68,25 @@ class AddPictureActivity : AppCompatActivity() {
         }
     }
 
+    private fun checkIntentExtras(extras : Bundle?){
+        if (extras != null && extras["categories"] is ArrayList<*>) {
+            val givenCategories = extras["categories"] as ArrayList<Category>
+            categories = categories.plus(givenCategories)
```

Comment:
Because givenCategories is an ArrayList and we can not write Set = ArrayList ^^' and i Thought it was the easiest way of doing. Tell me if you think smth else is better :)